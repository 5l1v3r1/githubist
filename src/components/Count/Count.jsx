// @flow

import React from 'react';

import styles from './Count.scss';

type Props = {
  count: number,
  title: string,
};

const Count = ({ count, title }: Props) => (
  <div className={styles.container}>
    <h3 className={styles.count}>{count.toLocaleString()}</h3>
    <p className={styles.title}>{title}</p>
  </div>
);

export default Count;
