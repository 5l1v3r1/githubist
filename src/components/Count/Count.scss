@import '../../theme/theme';

.container {
  padding: map-get($spacers, normal);
  border: 1px solid $text-muted;
  background-color: $white;
  border-radius: px-to-rem(2px);
  box-shadow: 3px 3px 0 #d5d5d6;
  text-align: center;
}

.count {
  margin: 0;
  color: $black-color;
  font-size: map-get($heading-sizes, h2);
  font-weight: 600;
}

.title {
  margin: 0 0 map-get($spacers, small);
  color: $text-muted;
  font-size: map-get($font-sizes, large);
}
