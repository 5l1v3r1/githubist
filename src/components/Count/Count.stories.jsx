// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import Count from './Count';

storiesOf('Count', module).add('default', () => <Count count={3250} title="GeliÅŸtirici" />);
