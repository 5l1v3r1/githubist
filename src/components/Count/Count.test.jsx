// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Count from './Count';

it('renders correctly', () => {
  const wrapper = shallow(<Count count={3250} title="GeliÅŸtirici" />);

  expect(wrapper).toMatchSnapshot();
});
