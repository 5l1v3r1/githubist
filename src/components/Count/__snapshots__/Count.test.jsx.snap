// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    3,250
  </h3>
  <p>
    Geli≈ütirici
  </p>
</div>
`;
